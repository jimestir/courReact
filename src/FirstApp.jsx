import React from 'react'
import PropTypes from 'prop-types'

function FirstApp ({ title, Subtitle, numberPage }) {
  return (
    <>
      <h1> {title} </h1>
      <h2> {Subtitle} </h2>
      <span> {numberPage} </span>
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  Subtitle: PropTypes.string,
  numberPage: PropTypes.number.isRequired
}
FirstApp.defaultProps = {
  Subtitle: 'There is no title'
}
export default FirstApp
