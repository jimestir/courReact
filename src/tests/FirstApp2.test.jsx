import { render, screen } from '@testing-library/react'
import FirstApp from '../FirstApp'

describe('test in <FirstApp />', () => {
  const title = 'Hello i am Jimmy'
  const numberPage = 24
  const subTitle = 'soy un subtitulo'

  test('Debe hacer match con el snapshot', () => {
    const { container } = render(<FirstApp title={title} numberPage={numberPage} />)
    expect(container).toMatchSnapshot()
  })
  test('Debe de mostrar el mensaje "Hello i am Jimmy"', () => {
    render(<FirstApp title={title} numberPage={numberPage} />)
    // screen.debug()
    expect(screen.getByText(title)).toBeTruthy()
  })
  test('Debe de mostrar el titulo en un h1', () => {
    render(<FirstApp title={title} numberPage={numberPage} />)
    expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title)
  })
  test('Debe de mostar el subtitulo enviado por props', () => {
    render(<FirstApp title={title} Subtitle={subTitle} numberPage={numberPage} />)
    expect(screen.getAllByText(subTitle).length).toBe(3)
  })
})
