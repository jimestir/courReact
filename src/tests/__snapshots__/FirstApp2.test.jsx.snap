// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in <FirstApp /> Debe hacer match con el snapshot 1`] = `
<div>
  <h1
    class="h1"
    data-testid="test-title"
  >
     
    Hello i am Jimmy
     
  </h1>
  <h2>
    There is no title
  </h2>
  <h2>
    There is no title
  </h2>
  <h2>
    There is no title
  </h2>
  <span>
    24
  </span>
</div>
`;
